web3/contract.py

```diff
@@ -1474,19 +1476,64 @@ def processLog(self, log: HexStr) -> EventData:
         return get_event_data(self.w3.codec, self.abi, log)
 
     @combomethod
-    def create_filter(
+    def _get_event_filter_params(
+        self,
+        abi: ABIEvent,
+        argument_filters: Optional[Dict[str, Any]] = None,
+        fromBlock: Optional[BlockIdentifier] = None,
+        toBlock: Optional[BlockIdentifier] = None,
+        blockHash: Optional[HexBytes] = None,
+    ) -> FilterParams:
+
+        if not self.address:
+            raise TypeError(
+                "This method can be only called on "
+                "an instated contract with an address"
```

Comment:
```suggestion
                "an instantiated contract with an address"
```

Unrelated to these changes but I think this is a typo?

Reply:
yepper!